install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -835,10 +834,8 @@ data:
   enable-k8s-event-handover: "true"
 {{- end }}
 
-{{- if hasKey .Values "crdWaitTimeout" }}
-  crd-wait-timeout: {{ .Values.crdWaitTimeout | quote }}
-{{- else if ( ne $crdWaitTimeout "5m" ) }}
-  crd-wait-timeout: {{ $crdWaitTimeout | quote }}
+{{- if .Values.crdWaitTimeout }}
```

Comment:
I have a small concern that changing the `if hasKey ...` to `if ...` will cause these settings to be auto-enabled for older clusters during upgrade rather than using cilium-agent daemon flags defaults (as defined in `daemon/cmd/...`).

I think we could check this by running `helm template ...` and looking at the configmap before this commit, then comparing it to `helm template ...` and look at the configmap after this commit? I've done some manual upgrade testing like this in the past, but I never got a chance to turn it into some automated upgrade test.

Reply:
I did helm template before and after the commit when we first started this helm documentation and cleanup PRs, the `if haskey ... ` to `if ... ` does not add to the configmap if the helm option is not set

Reply:
OK yep that makes sense, and with the values file specifying `""` as the option, this should evaluate false. :pray: